backend/src/core/middlewares/api-key.middleware.ts

```diff
@@ -0,0 +1,50 @@
+import { loggerWithLabel } from '@core/logger'
+import { Handler, NextFunction, Request, Response } from 'express'
+import { ApiKeyService, AuthService } from '@core/services'
+
+export interface ApiKeyMiddleware {
+  deleteApiKey: Handler
+}
+export const InitApiKeyMiddleware = (authService: AuthService) => {
+  const logger = loggerWithLabel(module)
+
+  const deleteApiKey = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<Response | void> => {
+    try {
+      // enforce authentication by cookie
+      const authenticated = authService.checkCookie(req)
+      if (!authenticated) {
+        return res.sendStatus(403).json({
+          message: `Invalid cookie`,
+        })
+      }
+
+      const { apiKeyId } = req.params
+      const userId = req.session?.user?.id
+
+      const count = await ApiKeyService.deleteApiKey(
+        userId.toString(),
+        +apiKeyId
+      )
+      if (count <= 0) {
```

Comment:
**[question/nit]**
- can `count` be less than zero?
- took a look at the API, seems like it returns the number of destroyed rows, maybe would be good to rename the variable to something clearer

Reply:
renamed